completions.go

```diff
@@ -1,34 +1,88 @@
 package kingpin
 
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"strings"
+)
+
+type completion struct {
```

Comment:
What is the purpose of splitting this into two separate lists? Can you document this?

As mentioned above, it will also need to be public and usable by API consumers for completion. So either helper functions or fields will need to be made public too.